### PR TITLE
feat(src): add src prop for reverse proxy

### PR DESCRIPTION
**Pull request by tooppaaa**
Issue: #22 

I also changed implementation to use FunctionalComponent.

I let you review and tests, I'm not sur how you handle testing on this feature :)

### CODE REVIEW

**Comment by jawadst** [comment]
@tooppaaa 
Thanks a lot for sending a PR for this. I'll let @MaximePlancke review.

That being said and even if we merge this one: on the `src`, simply changing it here is not enough as our SDK also loads other files that are hard coded to `sdk.privacy-center.org`. We'll need to make an update to the Core SDK to allow configuring that value (which is planned on our side).

**Comment by tooppaaa** [comment]
Hope it helps, feel free to drop it if that's not going in your direction !

If I understand correctly - the self hosting feature documented isn't ready yet ? We can continue that discussions by mail as it's not related to this open source project.

File: src/index.js [hunk 1]
```diff
@@ -1,207 +1,206 @@
...
     }
   }
 
-  componentDidMount() {
-    this.init();
+  React.useEffect(() => {
```
Comment (MaximePlancke) [inline src/index.js:169]: Hooks started to be available after version `16.8`. Even though we use a `^version` in our package, I would suggest we upgrade the version number from `^16.5.2` to `^16.8`

File: src/index.js [hunk 2]
```diff
@@ -1,207 +1,206 @@
...
+  onPreferencesClickAgreeToAll =  () => {},
+  onPreferencesClickDisagreeToAll =  () => {},
+  onPreferencesClickPurposeAgree =  () => {},
+  onPreferencesClickPurposeDisagree =  () => {},
+  onPreferencesClickViewVendors =  () => {},
+  onPreferencesClickSaveChoices =  () => {},
+  onPreferencesClickVendorAgree =  () => {},
+  onPreferencesClickVendorDisagree =  () => {},
+  onPreferencesClickVendorSaveChoices =  () => {},
+  src = "https://sdk.privacy-center.org/"
```
Comment (MaximePlancke) [inline src/index.js:26]: We should also update the documentation with this new `prop`

File: src/index.js [hunk 3]
```diff
@@ -1,207 +1,206 @@
-import { Component } from 'react';
+import React from 'react';
```
Comment (MaximePlancke) [inline src/index.js:1]: We don't have automated test yet on this project but if you want to do manual tests, you can run `npm run start` and play with the file located in `demo/src/index.js` by adding a `src` prop

**Review by MaximePlancke — COMMENTED** [review]
Hello @tooppaaa , thank you for your PR!

I added some comments. As @jawadst mentioned is it not an option available in our SDK yet and merging this could result in confusion but it looks good and will definitely merge it once the feature is fully available. We will probably also need to add this `src` option in the config object so it can be passed to the SDK.

File: src/index.js (hunk 2 above)
Comment (jawadst) [reply src/index.js:26]: Done in new MR

File: src/index.js (hunk 3 above)
Comment (jawadst) [reply src/index.js:1]: I have updated the tests in a new MR

File: src/index.js (hunk 1 above)
Comment (jawadst) [reply src/index.js:169]: Done in a new MR

**Comment by jawadst** [comment]
@tooppaaa I needed to make a few changes so I am closing this MR in favor of https://github.com/didomi/react/pull/25/files that includes your commit and additional tests.

We are planning a release of our Web SDK next week (week of 11/16) that will unblock this MR. Once it's deployed, I'll ping you and you'll be able to start using the new `sdkPath` property.

**Comment by tooppaaa** [comment]
Sorry for not picking it up, thanks a lot for this !

**Pull request closed without merging**